### DISCUSSION TITLE
GET with a body or caching POSTs

### DISCUSSION THREAD

**Discussion by aarondfrancis** [vercel/community#292] [Help] [answered]
Hey friends 👋 

I have a question mostly about caching. I'm working on a tool that does some syntax highlighting via API (https://torchlight.dev) and I'd like to take advantage of caching, but I can't figure out how to do that within Vercel's platform.

As of now, the users POSTs a request with the code block's content, the theme, and some options. I parse it, highlight it, and send it back.

I'd like to implement HTTP caching, but Vercel's docs state that the request must be a GET or HEAD.

So I tried sending a body with a GET request, [like elastic does](https://www.elastic.co/guide/en/elasticsearch/guide/current/_empty_search.html), but the body didn't make it through.

In my ideal world the Torchlight clients would either POST or GET to `https://api.torchlight.dev/highlight/[unique block hash]` and I could take advantage of HTTP caching.

TL;DR: I can't cache POSTs and I can't send bodies with GETs.

Any ways to work around this?

Thanks!

**Comment by aarondfrancis** [comment, marked as answer]
Ok update, it looks like the tool I was using wasn't sending the request body with GET, so that part is solved.

Now I'm getting a header of `x-vercel-cache: BYPASS` even though I've set `cache-control: s-maxage=60`.

Any ideas why it's getting bypassed?

**Reply by crowterli** [reply]
Hi Aaron,

With regards to the bypass, it may be useful if you can share [a HAR file showing this behaviour](https://toolbox.googleapps.com/apps/har_analyzer/).

Certainly, within the context of Vercel, [we require that the caching can only be done on a GET or HEAD request.](https://vercel.com/docs/concepts/edge-network/caching#cacheable-responses). I don't think there's any getting around this requirement.

From a more general standpoint, my understanding is that it _is_ possible to cache POST requests, but to do so is a bit of a juxtaposition from what POST's are intended to do, since their purpose is to facilitate data submission and caching that is not really sensible.

Thanks,
Liam
